modules/combined/examples/geochem-porous_flow/geotes_2D/porous_flow.i

```diff
@@ -201,18 +201,22 @@ production_rate = 1.0 # kg/s/m
   [mole_rate_Na_produced]
     type = FunctionValuePostprocessor
     function = moles_Na
+    indirect_dependencies = 'kg_Na_produced_this_timestep dt'
```

Comment:
The issue is that commonly parsed functions depend on other parsed functions and those parsed functions may depend on user objects. It is pretty difficult to catch all the dependencies automatically without an arbitrary dependency scheme. Without that we would have to add function dependencies to the `FunctionInterface` and so when a function depends on another function we could recurse through the function dependencies and add any user object dependencies of the *functions* to the original user object which depends on a function(s)

Reply:
Oh, this is more difficult than I imagined. Since this is possibly the only postprocessor requiring that, I guess it is fine to ask users to manually input those depending uos/pps. I do not like this but if this is too difficult to do, it will be fine to me to add an alert in its markdown to say that `indirect_dependencies` must be set in cases like here.

Reply:
Um, I thought I had added postprocessor dependency pass through to functions at some point...

Reply:
If you did it doesn't work :-/

Reply:
Oh, I was thinking of this: https://github.com/idaholab/moose/issues/16354